metis-media-service/src/main/java/eu/europeana/metis/mediaservice/MediaProcessor.java

```diff
@@ -0,0 +1,466 @@
+			}
+		} catch (MediaException e) {
+			throw e;
+		} catch (Exception e) {
+			LOGGER.info("Could not parse ffprobe response:\n" + StringUtils.join(resultLines, "\n"), e);
+			throw new MediaException("File seems to be corrupted", "AUDIOVIDEO ERROR");
+		}
+	}
+	
+	private JSONObject findStream(JSONObject data, String codecType) {
+		for (Object streamObject : data.getJSONArray("streams")) {
+			JSONObject stream = (JSONObject) streamObject;
+			if (codecType.equals(stream.getString("codec_type")))
+				return stream;
+		}
+		return null;
+	}
+	
+	private void processText(String url, String mimeType, File contents) throws IOException {
+		if (!shouldExtractMetadata(urlTypes.get(url)))
+			return;
+		
+		boolean containsText = mimeType.startsWith("text/");
+		Integer resolution = null;
+		
+		if ("application/pdf".equals(mimeType)) {
+			PdfReader reader = new PdfReader(contents.getAbsolutePath());
+			try {
+				PdfReaderContentParser parser = new PdfReaderContentParser(reader);
+				PdfListener pdfListener = new PdfListener();
+				for (int i = 1; i <= reader.getNumberOfPages(); i++) {
+					parser.processContent(i, pdfListener);
+					resolution = pdfListener.dpi;
+					containsText = !StringUtils.isBlank(pdfListener.getResultantText());
+					if (resolution != null && containsText)
+						break;
+				}
+			} finally {
+				reader.close();
+			}
+		}
+		
+		WebResource resource = edm.getWebResource(url);
+		resource.setMimeType(mimeType);
+		resource.setFileSize(contents.length());
+		resource.setContainsText(containsText);
+		resource.setResolution(resolution);
+	}
+	
+	List<String> runCommand(List<String> command, boolean mergeError) throws IOException {
+		return runCommand(command, mergeError, null);
+	}
+	
+	List<String> runCommand(List<String> command, boolean mergeError, byte[] inputBytes) throws IOException {
+		Process process = new ProcessBuilder(command).redirectErrorStream(mergeError).start();
+		if (!mergeError) {
+			commandIOThreadPool.execute(() -> {
+				try (InputStream errorStream = process.getErrorStream()) {
+					String output = IOUtils.toString(errorStream, Charset.defaultCharset());
+					if (!StringUtils.isBlank(output))
+						LOGGER.warn("Command: {}\nerror output:\n{}", command, output);
+				} catch (IOException e) {
+					LOGGER.error("Error stream reading faild for command " + command, e);
+				}
+			});
+		}
+		if (inputBytes != null) {
+			commandIOThreadPool.execute(() -> {
+				try (OutputStream processInput = process.getOutputStream()) {
+					processInput.write(inputBytes);
+				} catch (IOException e) {
+					LOGGER.error("Pushing data to process input stream failed for command " + command, e);
+				}
+			});
+		}
+		try (InputStream in = process.getInputStream()) {
+			return IOUtils.readLines(in, Charset.defaultCharset());
+		}
+	}
+	
+	/**
+	 * This is a non-static method because it needs access to
+	 * {@link #runCommand(List, boolean)}.
+	 */
```

Comment:
Why not make it static and pass an instance of MediaProcessor as argument? Seems a lot less hacky! Or even access to a thread pool, because that is the only thing why it needs an instance of MediaProcessor.